### PR TITLE
c2c: add acceptance test

### LINKED ISSUE #99230: c2c: write local c2c roachtest for CI
We often merge c2c features without double checking how it affects the c2c roachtest framework. To avoid this, let's write a local roachtest that runs through CI that exercises the c2c roachtest framework.

### PR DESCRIPTION
**Pull request by msbutler**
c2c: add acceptance test

This patch adds a lightweight acceptance test that calls the main c2c roachtest
driver. This will add a c2c roachtest to Essential CI, preventing future
patches from inadvertently breaking the C2C roachtest driver.

Fixes #99230

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/acceptance.go
Comment (stevendanna) [inline pkg/cmd/roachtest/tests/acceptance.go:75]: Do we need 3 nodes? Can we run the workload on the src node? I suppose since this is "local" it doesn't really matter?

**Review by stevendanna — APPROVED** [review]
Seems reasonable to me. Let's be ready to remove this quickly if we end up causing instability in CI.

How long does this test take to run for you locally? Has it been stable?

**Comment by msbutler** [comment]
I've run it locally a bunch of times and it's been stable. It take less than 5 minutes to run.

**Comment by renatolabs** [comment]
I started a 100-run of this test earlier today and haven't seen failures so far. Can we wait for this to finish before we merge? For comparison, when we recently merged a change that made `acceptance/version-upgrade` to flake very frequently in CI, it took 48 runs to reproduce it locally (in other words, CI has a special sauce that is difficult to replicate, so we should run it hundreds of times locally before we have any hopes of stability in CI).

It won't block the merge for too long (currently at 70 runs), I'll send an update when it finishes.

**Review by renatolabs — APPROVED** [review]
Ran 100 times without failures 🟢

**Pull request merged**
